keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -119,12 +119,24 @@ class CombinedRepository(
   }
 
   /**
-   * Deletes a delivery config and everything in it.
+   * Fetches delivery config by name, then deletes everything in it.
+   */
+  @Transactional(propagation = REQUIRED)
+  override fun deleteDeliveryConfig(deliveryConfigName: String): DeliveryConfig =
+    deleteDeliveryConfigFull(deliveryConfigRepository.get(deliveryConfigName))
+
+  /**
+   * Fetches delivery config by application, then deletes everything in it.
    */
-  @Transactional(propagation = Propagation.REQUIRED)
-  override fun deleteDeliveryConfig(deliveryConfigName: String): DeliveryConfig {
-    val deliveryConfig = deliveryConfigRepository.get(deliveryConfigName)
+  @Transactional(propagation = REQUIRED)
+  override fun deleteDeliveryConfigByApplication(application: String): DeliveryConfig =
+    deleteDeliveryConfigFull(deliveryConfigRepository.getByApplication(application))
 
+  /**
+   * Deletes a delivery config and everything in it.
+   */
+  @Transactional(propagation = REQUIRED)
+  internal fun deleteDeliveryConfigFull(deliveryConfig: DeliveryConfig): DeliveryConfig {
```

Comment:
Not related to this PR, but I think the code below could be a lot shorter if we used proper foreign keys in the database. We could just do a cascading delete.

Reply:
ha, that's part of a longstanding argument about efficiency and database design between asher and fletch. we could revisit, but we should do that by properly talking to sql experts and understanding the tradeoffs about how that scales